test/ng/compileSpec.js

```diff
@@ -294,6 +294,26 @@ describe('$compile', function() {
       inject(function($compile) {});
     });
 
+    it('should omit special chars before processing directive name', function() {
```

Comment:
If this applies only to directives with restrict: 'A', then the test description should reflect that. Maybe even link to this issue in a comment.